### PR TITLE
Fix scanning issues

### PR DESCRIPTION
**Pull request by matt-goldman**
This PR fixes the following issues:

 - Stop app crashing on successful scan
 - Reinitialise scanner on unsuccessful scan
 - Update profile on successful scan

Closes: https://github.com/SSWConsulting/SSW.Rewards/issues/289 and https://github.com/SSWConsulting/SSW.Rewards/issues/267

Additionally, resolves an issue with the coin animation (could not fixed, replaced with an alternative animation).

See: https://youtu.be/r3KoKHL25kg

### CODE REVIEW

File: Xamarin/SSW.Rewards/SSW.Rewards/Pages/ScanPage.xaml.cs [hunk 1]
```diff
@@ -9,29 +9,35 @@ namespace SSW.Rewards.Pages
     [XamlCompilation(XamlCompilationOptions.Compile)]
     public partial class ScanPage : ContentPage
     {
+        public const string EnableScannerMessage = "EnableScanner";
```
Comment (lukecookssw) [inline Xamarin/SSW.Rewards/SSW.Rewards/Pages/ScanPage.xaml.cs:12]: Should this be `private` ?

File: Xamarin/SSW.Rewards/SSW.Rewards/ViewModels/ScanResultViewModel.cs [hunk 2]
```diff
@@ -50,7 +59,7 @@ public async Task CheckScanData()
                     if(result.ScanType == ScanType.Achievement)
                     {
                         ResultHeading = "Achivement Added!";
```
Comment (lukecookssw) [inline Xamarin/SSW.Rewards/SSW.Rewards/ViewModels/ScanResultViewModel.cs:61]: Fix typo "achivement"

**Review by lukecookssw — CHANGES_REQUESTED** [review]

File: Xamarin/SSW.Rewards/SSW.Rewards/Pages/ScanPage.xaml.cs (hunk 1 above)
Comment (matt-goldman) [reply Xamarin/SSW.Rewards/SSW.Rewards/Pages/ScanPage.xaml.cs:12]: No. It's referenced elsewhere. See https://github.com/SSWConsulting/SSW.Rewards/issues/312 for resolving this.

File: Xamarin/SSW.Rewards/SSW.Rewards/ViewModels/ScanResultViewModel.cs (hunk 2 above)
Comment (matt-goldman) [reply Xamarin/SSW.Rewards/SSW.Rewards/ViewModels/ScanResultViewModel.cs:61]: Nice catch, done.

**Review by lukecookssw — APPROVED** [review]
👍

**Pull request merged**